packages/device-registry/src/blockchain-facade/ProducingDevice.ts

```diff
@@ -100,14 +100,7 @@ export class Entity extends Device.Entity implements IProducingDevice {
     }
 
     get lastSmartMeterReadWh(): number {
-        const smReads = this.offChainProperties.smartMeterReads;
-
-        if (smReads.length < 1) {
-            return 0;
-        }
-
-        const lastSmReadIndex = this.offChainProperties.smartMeterReads.length - 1;
-        return this.offChainProperties.smartMeterReads[lastSmReadIndex].meterReading;
+        return this.offChainProperties.lastSmartMeterReading?.meterReading ?? 0;
```

Comment:
I think it should return `null` instead of `0` in case reads aren't loaded. Otherwise how to tell the difference?